ReactAndroid/src/main/java/com/facebook/react/views/swiperefresh/ReactSwipeRefreshLayout.java

```diff
@@ -20,11 +20,26 @@
  */
 public class ReactSwipeRefreshLayout extends SwipeRefreshLayout {
 
+  private boolean mRefreshing = false;
+
   public ReactSwipeRefreshLayout(ReactContext reactContext) {
     super(reactContext);
   }
 
   @Override
+  public void setRefreshing(boolean refreshing) {
+    mRefreshing = refreshing;
```

Comment:
Yes, the problem is when setRefreshing gets called twice in the same bridge transaction the order the runnables get executed is bad, see the example in the PR desc, adding the field makes sure is uses the latest value.